drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -2858,12 +2858,40 @@ static int vc4_hdmi_audio_init(struct vc4_hdmi *vc4_hdmi)
 
 }
 
-static irqreturn_t vc4_hdmi_hpd_irq_thread(int irq, void *priv)
+static void vc4_hdmi_hpd_con_wq(struct work_struct *work)
+{
+	struct vc4_hdmi *vc4_hdmi = container_of(to_delayed_work(work),
+						 struct vc4_hdmi,
+						 hpd_con_work);
+	struct drm_connector *connector = &vc4_hdmi->connector;
+	struct drm_device *dev = connector->dev;
+
+	if (dev && dev->registered)
+		drm_connector_helper_hpd_irq_event(connector);
+}
+
+#define HPD_DEBOUNCE_DELAY_MS 2500
```

Comment:
Please dont put defines here instead place it near the top of the file

```diff
- #define HPD_DEBOUNCE_DELAY_MS 2500
```

You can put it near line 121

```diff
#define HSM_MIN_CLOCK_FREQ	120000000
#define CEC_CLOCK_FREQ 40000

#define HDMI_14_MAX_TMDS_CLK   (340 * 1000 * 1000)
+ #define HPD_DEBOUNCE_DELAY_MS 2500
/* bit field to force hotplug detection. bit0 = HDMI0 */
static int force_hotplug = 0;
module_param(force_hotplug, int, 0644);

Reply:
Who are you?

Reply:
Why?

Reply:
I'm only suggesting changes.